src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -993,26 +987,63 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
                 "Response version " + responseVersion);
         }
 
-        private async Task<Version> SendRequestAndGetResponseVersionAsync(Version requestVersion, Uri server)
+        private async Task<Version> SendRequestAndGetResponseVersionAsync(Version requestVersion, Uri serverUri)
         {
-            var request = new HttpRequestMessage(HttpMethod.Get, server);
-            if (requestVersion != null)
-            {
-                request.Version = requestVersion;
-            }
-            else
+            Version responseVersion = null;
+
+            if (requestVersion == null)
             {
                 // The default value for HttpRequestMessage.Version is Version(1,1).
                 // So, we need to set something different to test the "unknown" version.
-                request.Version = new Version(0,0);
+                requestVersion = new Version(0,0);
             }
+            _output.WriteLine("Request version: {0}", requestVersion);
 
-            using (var client = new HttpClient())
-            using (HttpResponseMessage response = await client.SendAsync(request))
+            if (serverUri == null)
             {
-                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
-                return response.Version;
+                await LoopbackServer.CreateClientAndServerAsync(async (handler, client, server, url) =>
```

Comment:
@CIPop Yes, the LoopbackServer could be refactored to provide better methods.  But this PR was focused on using it as-is to help stabilize these particular tests.